dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -2037,28 +2054,25 @@ def dpnp_matmul(
         # TODO: investigate usage of syrk function from BLAS in
         # case of a.T @ a and a @ a.T to gain performance.
         if x1_is_2D and x2_is_2D:
-            ht_blas_ev, _ = bi._gemm(
+            ht_blas_ev, _, row_major = bi._gemm(
                 exec_q,
                 dpnp.get_usm_ndarray(x1),
                 dpnp.get_usm_ndarray(x2),
                 dpnp.get_usm_ndarray(result),
                 dep_events_list,
             )
+            host_tasks_list.append(ht_blas_ev)
+            if not row_major:
+                result = dpnp.reshape(result.ravel(), result.shape, order="F")
```

Comment:
Can we allocate the result with proper contig layout to avoid extra memory copying?

Reply:
It is might be possible, I will try to implement it in another PR.